components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilTest.java

```diff
@@ -18,18 +18,206 @@
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.sso.saml.SAMLSSOConstants;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.tenant.TenantManager;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.*;
+
+/**
+ * Unit test cases for SAMLSSOUtil.
+ */
+@PrepareForTest(IdentityProviderManager.class)
 public class SAMLSSOUtilTest {
-    @org.testng.annotations.Test
+
+    @Mock
+    private RealmService realmService;
+
+    @Mock
+    private TenantManager tenantManager;
+
+    @Mock
+    private IdentityProviderManager identityProviderManager;
+
+    @Mock
+    private IdentityProvider identityProvider;
+
+    @Mock
+    private FederatedAuthenticatorConfig federatedAuthenticatorConfig;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeTest
+    public void initTest() throws UserStoreException, IdentityProviderManagementException {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    private void prepareForGetIssuer() throws UserStoreException,
+            IdentityProviderManagementException {
+
+        when(tenantManager.getTenantId(anyString())).thenReturn(-1234);
+        when(realmService.getTenantManager()).thenReturn(tenantManager);
+
+        SAMLSSOUtil.setRealmService(realmService);
+
+        Property property = new Property();
+        property.setName("IdPEntityId");
+        property.setValue(TestConstants.LOACALHOST_DOMAIN);
+        Property[] properties = {property};
+        when(federatedAuthenticatorConfig.getProperties()).thenReturn(properties);
+        when(federatedAuthenticatorConfig.getName()).thenReturn(
+                IdentityApplicationConstants.Authenticator.SAML2SSO.NAME);
+        FederatedAuthenticatorConfig[] fedAuthConfs = {federatedAuthenticatorConfig};
+        when(identityProvider.getFederatedAuthenticatorConfigs()).thenReturn(fedAuthConfs);
+
+        mockStatic(IdentityProviderManager.class);
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
+        when(identityProviderManager.getResidentIdP(anyString())).thenReturn(identityProvider);
+    }
+
+    @Test
+    public void testGetIssuer() throws Exception {
+
+        prepareForGetIssuer();
+        Issuer issuer = SAMLSSOUtil.getIssuer();
+        assertEquals(issuer.getValue(), TestConstants.LOACALHOST_DOMAIN, "Super tenant mode.");
```

Comment:
What does super tenant domain means when the test is failed ? I think this is not really meaningful